### DISCUSSION TITLE
CompileError: Could not find declaration "runApp" on flutter 3.3.0

### DISCUSSION THREAD

**Discussion by zmtzawqlp** [ethanblake4/flutter_eval#9] [General]
i am follow the readme, add flutter_eval.json into project and run `dart_eval compile -o program.evc`, get some error, would you please check it if you have time, thanks.

[demo.zip](https://github.com/ethanblake4/flutter_eval/files/9525457/demo.zip)

```
CompileError: Could not find declaration "runApp"
#0      IdentifierReference.getValue (package:dart_eval/src/eval/compiler/reference.dart:199:57)
#1      compileIdentifier (package:dart_eval/src/eval/compiler/expression/identifier.dart:13:48)
#2      compileMethodInvocation (package:dart_eval/src/eval/compiler/expression/method_invocation.dart:30:20)
#3      compileExpression (package:dart_eval/src/eval/compiler/expression/expression.dart:28:12)
#4      compileExpressionAndDiscardResult (package:dart_eval/src/eval/compiler/expression/expression.dart:76:5)
#5      compileStatement (package:dart_eval/src/eval/compiler/statement/statement.dart:21:5)
#6      compileBlock (package:dart_eval/src/eval/compiler/statement/block.dart:16:20)
#7      compileFunctionDeclaration (package:dart_eval/src/eval/compiler/declaration/function.dart:50:14)
#8      compileDeclaration (package:dart_eval/src/eval/compiler/declaration/declaration.dart:20:5)
#9      Compiler.compileSources.<anonymous closure>.<anonymous closure> (package:dart_eval/src/eval/compiler/compiler.dart:301:9)
#10     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:617:13)
#11     Compiler.compileSources.<anonymous closure> (package:dart_eval/src/eval/compiler/compiler.dart:290:13)
#12     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:617:13)
#13     Compiler.compileSources (package:dart_eval/src/eval/compiler/compiler.dart:287:30)
#14     Compiler.compile (package:dart_eval/src/eval/compiler/compiler.dart:117:12)
#15     main (file:///Users/zmt/.pub-cache/hosted/pub.flutter-io.cn/dart_eval-0.4.5/bin/dart_eval.dart:103:36)
#16     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:32)
#17     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
```

**Comment by ethanblake4** [comment]
Hi, flutter_eval does not support `runApp`. That's because you are supposed to use it inside of an existing Flutter app, which is already calling `runApp`. So, you can just comment out the runApp part, or even the entire `main` function. Instead of calling `main` in EvalWidget, you should specify a widget constructor, like "MyApp." (don't forget the `.`)

**Comment by zmtzawqlp** [comment]
> u should specify a widget constructo

thanks for quick reply, it's better to add those comments to readme, and give the user a full successful steps. thanks

**Comment by ethanblake4** [comment]
Added documentation to readme in https://github.com/ethanblake4/flutter_eval/commit/ccb607ef6932dc60139e94fe8e27cd48c4589135. Closing

**Comment by zmtzawqlp** [comment]
still has some error. after i change as readme said:

``` dart
Widget main() {
  // do any setup, then
  return MyApp();
}
// void main() {
//   //runApp(const MyApp());
// }
```

dart_eval: 0.4.7

``` dart
Unhandled exception:
Null check operator used on a null value
#0      _resolveImportsAndExports.<anonymous closure> (package:dart_eval/src/eval/compiler/compiler.dart:620:87)
#1      MappedListIterable.elementAt (dart:_internal/iterable.dart:413:31)
#2      ListIterable.toSet (dart:_internal/iterable.dart:218:18)
#3      _resolveImportsAndExports (package:dart_eval/src/eval/compiler/compiler.dart:620:90)
#4      Compiler.compileSources (package:dart_eval/src/eval/compiler/compiler.dart:165:9)
#5      Compiler.compile (package:dart_eval/src/eval/compiler/compiler.dart:118:12)
#6      main (file:///Users/zmtzawqlp/Documents/github/hot_update/flutter_eval/dart_eval/bin/dart_eval.dart:103:36)
#7      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:32)
#8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
```

**Reply by ethanblake4** [reply]
@zmtzawqlp This probably means you are `import`ing something that doesn't exist in flutter_eval. Flutter_eval does not have all of the imports available to Dart/Flutter yet. Unfortunately the error message is quite unhelpful so I've filed https://github.com/ethanblake4/dart_eval/issues/47 to show exactly what import is failing, I think this can be released in dart_eval v0.5 which will be out soon.

In the meantime, you can try removing some of your imports to see what causes it. I'll also remind you that dart_eval does not support compiling packages in `pubspec.yaml` yet.

**Reply by zmtzawqlp** [reply]
by the way, some error in flutter_eval demo.

'dart:typed_data' not need in 3.3.0. but it's needed for 3.0.5.  flutter_eval require flutter: ">=3.0.0 <3.4.0"

/Users/zmtzawqlp/Documents/flutter/3.0.0/.pub-cache/hosted/pub.flutter-io.cn/flutter_eval-0.4.7/lib/src/flutter_eval.dart

**Reply by zmtzawqlp** [reply]
> @zmtzawqlp This probably means you are `import`ing something that doesn't exist in flutter_eval. Flutter_eval does not have all of the imports available to Dart/Flutter yet. Unfortunately the error message is quite unhelpful so I've filed [ethanblake4/dart_eval#47](https://github.com/ethanblake4/dart_eval/issues/47) to show exactly what import is failing, I think this can be released in dart_eval v0.5 which will be out soon.
> 
> In the meantime, you can try removing some of your imports to see what causes it. I'll also remind you that dart_eval does not support compiling packages in `pubspec.yaml` yet.

my import is only add flutter_eval, no any other import in the demo

**Reply by ethanblake4** [reply]
> my import is only add flutter_eval, no any other import in the demo

You cannot import flutter_eval in a project you are compiling with flutter_eval, that is like asking flutter_eval to compile itself (which it definitely can't do yet)

The only thing you should import is Flutter (or Dart) libraries.

**Reply by ethanblake4** [reply]
> by the way, some error in flutter_eval demo.

thanks, I filed https://github.com/ethanblake4/flutter_eval/issues/10 and will take a look

**Reply by zmtzawqlp** [reply]
> > my import is only add flutter_eval, no any other import in the demo
> 
> You cannot import flutter_eval in a project you are compiling with flutter_eval, that is like asking flutter_eval to compile itself (which it definitely can't do yet)
> 
> The only thing you should import is Flutter (or Dart) libraries.

i have remove all of dependencies in project(it just a simple demo create be flutter create command), the error is still here

**Reply by ethanblake4** [reply]
> i have remove all of dependencies in project(it just a simple demo create be flutter create command), the error is still here

I can't reproduce this, are you sure it's the same error?

The default `flutter create` project does use some things that flutter_eval isn't compatible with like tear-offs, but it shouldn't have an import error unless you forgot to add the `.dart_eval/bindings/flutter_eval.json`. Make sure you did that, then you can use [this version](https://gist.github.com/ethanblake4/855fe4379c07be10e5313876e58348b6) of the Flutter counter template to get rid of the remaining errors.

**Reply by zmtzawqlp** [reply]
> > i have remove all of dependencies in project(it just a simple demo create be flutter create command), the error is still here
> 
> I can't reproduce this, are you sure it's the same error?
> 
> The default `flutter create` project does use some things that flutter_eval isn't compatible with like tear-offs, but it shouldn't have an import error unless you forgot to add the `.dart_eval/bindings/flutter_eval.json`. Make sure you did that, then you can use [this version](https://gist.github.com/ethanblake4/855fe4379c07be10e5313876e58348b6) of the Flutter counter template to get rid of the remaining errors.

ok, it creates program.evc after use [this version](https://gist.github.com/ethanblake4/855fe4379c07be10e5313876e58348b6)  thanks for your reply

**Reply by ethanblake4** [reply]
Issue fixed in v0.5

**Comment by zmtzawqlp** [comment]
may be we can provide annotation to define which class should be scan to generate program.evc, and one Eval() one .evc, so that we can decide which page/widget we need to hot update.

``` dart
@Eval(fileName:'HomePage.evc', library:"package:hot_update_demo/home_page.dart", function:"MyHomePage.")
class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() {
    return _MyHomePageState();
  }
}

```

**Reply by ethanblake4** [reply]
I've thought about this before, but what do you need it for? Is the size of the generated .evc too large?

**Reply by zmtzawqlp** [reply]
we can decide which class(route) has dynamic update, and decide whether use RuntimeWidget

``` dart
    return MaterialApp(
      title: 'Flutter Demo',
      onGenerateRoute: (setting) {
        return MaterialPageRoute(builder: (b) {
          return false
              ? MyHomePage()
              : RuntimeWidget(
                  uri: Uri.parse(
                      'asset:packages/hot_update_demo/assets/program.evc'),
                  library: 'package:hot_update_demo/home_page.dart',
                  function: 'MyHomePage.',
                  //args: [$String('Jessica'), null, null]),
                );
        });
      },
    );
```

**Reply by ethanblake4** [reply]
That part is already a feature request [here](https://github.com/ethanblake4/flutter_eval/issues/3). I am planning to do it :)

I don't think it should generate a separate .evc for each Eval() though, that would make it slower and take more space.

**Comment by zmtzawqlp** [comment]
it seems that dart_eval compiler not use AnalysisContextCollection to scan dart sdk and flutter sdk to generate the template code. maybe we can create a tool to do this , and reduce adaptation problems.

https://stackoverflow.com/questions/69670345/in-dart-how-do-i-get-from-the-ast-to-elements

and i made a plugin https://pub.dev/packages/ff_annotation_route to generate route, we can get dart type and other info, make a tool. 

I'm trying to understand the principle  of dart_eval and flutter_eval ,  does it support any logic (in theory)?  widget map is easy to do, we can scan dart sdk, flutter sdk, packages and generate the map for all the widget, type, class, enum. but i can't get how to support the logic? it's kind of you if you can clarify it when you have a time. thanks.

**Reply by ethanblake4** [reply]
> scan dart sdk and flutter sdk to generate the template code. maybe we can create a tool to do this, and reduce adaptation problems.

This is a goal as well, see https://github.com/ethanblake4/dart_eval/issues/20. Are you saying that you are interested in contributing to help with this? 

> I'm trying to understand the principle of dart_eval and flutter_eval

There's a little bit of info [here](https://pub.dev/packages/dart_eval#how-does-it-work), but dart_eval is basically a Dart compiler that compiles to a custom bytecode format (.evc) and a runtime that can run that EVC bytecode. It has very extensive [interop](https://pub.dev/packages/dart_eval#interop) so you can use classes/functions that were created outside dart_eval inside it (with a wrapper), and even extend those classes (with a bridge).

> how to support the logic?

You can look at the [compiler code](https://github.com/ethanblake4/dart_eval/tree/master/lib/src/eval/compiler) to see all of it, but here's a simple example. Let's say you write:

```dart
function myfunc() {
    var i = 2;
    i += 1;
    print(i);
}
```

that might compile to something like:
```dart
PushScope('F0:myfunc')
PushConstantInt(2) // var i = 2
PushConstantInt(1)
NumAdd(L0, L1) // i += 1
PushArg(L2)
InvokeExternal(0) // print()
Return(-1)
```

You can see what any code compiles to using the dart_eval CLI, by running `dart_eval dump program.evc` on an EVC file.

Then, once we have this generated, it is just an array of classes. For example, the NumAdd class is [implemented here](https://github.com/ethanblake4/dart_eval/blob/3cb310cba16f9dea0bbbabeed4a85d5eaff48fb7/lib/src/eval/runtime/ops/primitives.dart#L77). We [go through them in order](https://github.com/ethanblake4/dart_eval/blob/3cb310cba16f9dea0bbbabeed4a85d5eaff48fb7/lib/src/eval/runtime/runtime.dart#L458) and call `run()`.

**Reply by zmtzawqlp** [reply]
> For example

thanks for quickly reply.  yes, i'm interested in this project.  other project about code push, try to use dart2js and js2dart to support logic, but it has limitation and can't write as a normal dart/flutter .

That's what I expect is:

1. write as a normal dart/flutter project(as possible as we can)
2. tool to generate dynamic update code.
3. tool to generate sdk and unofficial packages code

I will take a deep look at into dart_eval/flutter_eval, see what I can do 🙂， by the way, is there any blog to introduce it？

**Reply by ethanblake4** [reply]
Sweet, I'm glad you want to help! Yes, those 1/2/3 are the goals.

The only 'blog posts' are the reddit posts I made ([first](https://www.reddit.com/r/dartlang/comments/na0l5h/introducing_dart_eval_an_extensible_dart/), [second](https://www.reddit.com/r/dartlang/comments/uchl6a/introducing_dart_eval_v02_the_dart_interpreter/), [third](https://www.reddit.com/r/FlutterDev/comments/wux1vf/introducing_flutter_eval_one_large_step_towards/)). It doesn't really show how to use it though.

Might be good to start by trying to bridge a simple Dart package on Pub, like your own [like_button](https://github.com/fluttercandies/like_button). You can see [here](https://github.com/ethanblake4/dart_eval/discussions/46) where I walked someone else through bridging their package. This way, you can learn how the bridging system works, so you can help to write the auto-generator for it :)

**Reply by zmtzawqlp** [reply]
ok, i will take a look at it, and try to understand it.

**Comment by zmtzawqlp** [comment]
```
════════ Exception caught by widgets library ═══════════════════════════════════
The following RuntimeException was thrown building RuntimeWidget(dirty, state: _RuntimeWidgetState#736bc):
dart_eval runtime exception: type 'Null' is not a subtype of type '$InstanceImpl' in type cast
#0      SetObjectPropertyImpl.run
package:dart_eval/…/ops/objects.dart:243
#1      Runtime.execute
package:dart_eval/…/runtime/runtime.dart:472
#2      Runtime.executeNamed
package:dart_eval/…/runtime/runtime.dart:463

RUNTIME STATE
=============
Program offset: 34210
Stack sample: [L0: Instance of 'BridgeSuperShim', L1: Instance of '$InstanceImpl', L2: null, L3: null, L4: null, L5: null, L6: null, L7: null, L8: null, L9: null]
Call stack: [0, -1]
TRACE:
34204: JumpIfNonNull (@34208 if L1 != null)
34205: PushConstant (C2)
34206: CopyValue (L1 <-- L2)
34207: Pop (1)
34208: PushBridgeSuperShim ()
34209: CreateClass (F38:"MyHomePage", super L2, vLen=1))
34210: SetObjectPropertyImpl (L3[0] = L1)  <<< EXCEPTION
34211: BridgeInstantiate (subclass L3, Ex#1))
34212: ParentBridgeSuperShim (shim L2, bridge L4)
34213: Return (L4)
```

**Comment by zmtzawqlp** [comment]
i make a simple demo , still has some error

[demo1.zip](https://github.com/ethanblake4/flutter_eval/files/9761138/demo1.zip)


```
════════ Exception caught by widgets library ═══════════════════════════════════
The following RuntimeException was thrown building RuntimeWidget(dirty, state: _RuntimeWidgetState#929fb):
dart_eval runtime exception: type '$InstanceImpl' is not a subtype of type 'BridgeSuperShim' in type cast
#0      ParentBridgeSuperShim.run
package:dart_eval/…/ops/bridge.dart:72
#1      Runtime.execute
package:dart_eval/…/runtime/runtime.dart:472
#2      Runtime.executeNamed
package:dart_eval/…/runtime/runtime.dart:463

RUNTIME STATE
=============
Program offset: 34215
Stack sample: [L0: Instance of 'BridgeSuperShim', L1: Instance of '$InstanceImpl', L2: $StatelessWidget$bridge, L3: null, L4: null, L5: null, L6: null, L7: null, L8: null, L9: null]
Call stack: [0, -1]
TRACE:
34209: Return (L7)
34210: Pop (1)
34211: PushScope (F37:378, 'MyApp.()')
34212: PushBridgeSuperShim ()
34213: CreateClass (F37:"MyApp", super L1, vLen=0))
34214: BridgeInstantiate (subclass L2, Ex#0))
34215: ParentBridgeSuperShim (shim L1, bridge L3)  <<< EXCEPTION
34216: Return (L3)
34217: PushScope (F37:779, 'MyHomePage.title (get)')
34218: PushObjectPropertyImpl (L0[0])
```

**Comment by zmtzawqlp** [comment]
it works after i add `args: [null],`,  it seems that we must be pass `args: [null],` if `MyApp({Key? key})` has one parameter

**Reply by ethanblake4** [reply]
Yep, explained [here](https://pub.dev/packages/flutter_eval#calling-functions-and-passing-arguments). Reason is for performance.

**Comment by zmtzawqlp** [comment]
Navigator and NavigatorState are support in flutter_eval, but it has some error when compile with dart_eval

``` dart
              NavigatorState navigatorState = Navigator.of(context);
              navigatorState.pushNamed('PageA');
```

full code:

```
import 'package:flutter/material.dart';

class HomePageWidget extends StatefulWidget {
  // ignore: use_key_in_widget_constructors
  const HomePageWidget();
  @override
  State<HomePageWidget> createState() => _HomePageWidgetState();
}

class _HomePageWidgetState extends State<HomePageWidget> {
  _HomePageWidgetState();
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Test'),
        actions: <Widget>[
          TextButton(
            child: Icon(Icons.add),
            onPressed: () {
              NavigatorState navigatorState = Navigator.of(context);
              navigatorState.pushNamed('PageA');
            },
          )
        ],
      ),
      body: Center(
          child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: <Widget>[
          const Text(
            'You have pushed the button this many times:',
          ),
          Text(
            '$_counter',
            style: Theme.of(context).textTheme.headline4,
          ),
        ],
      )),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}

```

error:

``` dart
Found binding file: .dart_eval/bindings/flutter_eval.json
Compiling package runtime_widgets...
Unhandled exception:
Null check operator used on a null value
#0      CompilerContext.lookupLocal (package:dart_eval/src/eval/compiler/context.dart:223:40)
#1      IdentifierReference.getValue (package:dart_eval/src/eval/compiler/reference.dart:163:23)
#2      compileIdentifier (package:dart_eval/src/eval/compiler/expression/identifier.dart:13:48)
#3      compileMethodInvocation (package:dart_eval/src/eval/compiler/expression/method_invocation.dart:30:20)
#4      compileExpression (package:dart_eval/src/eval/compiler/expression/expression.dart:30:12)
#5      compileReturn (package:dart_eval/src/eval/compiler/statement/return.dart:43:17)
#6      compileStatement (package:dart_eval/src/eval/compiler/statement/statement.dart:24:12)
#7      compileBlock (package:dart_eval/src/eval/compiler/statement/block.dart:16:20)
#8      compileMethodDeclaration (package:dart_eval/src/eval/compiler/declaration/method.dart:51:14)
#9      compileDeclaration (package:dart_eval/src/eval/compiler/declaration/declaration.dart:18:12)
#10     compileClassDeclaration (package:dart_eval/src/eval/compiler/declaration/class.dart:29:5)
#11     compileDeclaration (package:dart_eval/src/eval/compiler/declaration/declaration.dart:16:5)
#12     Compiler.compileSources.<anonymous closure>.<anonymous closure> (package:dart_eval/src/eval/compiler/compiler.dart:338:9)
#13     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:614:13)
#14     Compiler.compileSources.<anonymous closure> (package:dart_eval/src/eval/compiler/compiler.dart:327:13)
#15     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:614:13)
#16     Compiler.compileSources (package:dart_eval/src/eval/compiler/compiler.dart:324:30)
#17     Compiler.compile (package:dart_eval/src/eval/compiler/compiler.dart:131:12)
#18     main (file:///Users/zmtzawqlp/.pub-cache/hosted/pub.flutter-io.cn/dart_eval-0.5.0/bin/dart_eval.dart:103:36)
#19     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:32)
#20     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
```

**Reply by ethanblake4** [reply]
Fixed in v0.5.1

**Comment by zmtzawqlp** [comment]
isSelected and  selectedIcon properties of IconButton are not  in flutter 3.0.5 
```
  isSelected: args[16]?.$value ?? false,
      selectedIcon: args[17]?.$value,
```

**Reply by ethanblake4** [reply]
Fixed in v0.5.2

**Comment by zmtzawqlp** [comment]
``` dart
          TextButton(
            child: Icon(
              Icons.add,
              color: Colors.red,
            ),
            onPressed: () {
              Navigator.of(context).pushNamed(
                'PageA',
                arguments: <String, dynamic>{
                  'key': null,
                  'count': _counter,
                  'constructorName': '',
                  'argumentNames': ['key', 'count'],
                },
              );

            },
          )
        ],
      ),
```

error when i click button

``` 
════════ Exception caught by gesture ═══════════════════════════════════════════
The following RuntimeException was thrown while handling a gesture:
dart_eval runtime exception: type 'Null' is not a subtype of type '$Value'
#0      BoxMap.run
package:dart_eval/…/ops/primitives.dart:270
#1      Runtime.bridgeCall
package:dart_eval/…/runtime/runtime.dart:594
#2      EvalFunctionPtr.call
package:dart_eval/…/runtime/function.dart:52

RUNTIME STATE
=============
Program offset: 34371
Stack sample: [L0: [L0: $State$bridge<StatefulWidget>#4c604, L1: Instance of '$BuildContext', L2: null], L1: 1, L2: Instance of '$BuildContext', L3: Instance of '$NavigatorState', L4: $"PageA", L5: {$"key": null, $"count": 0, $"constructorName": $"", $"argumentNames": Instance of '$List<$Value>'}, L6: $"key", L7: null, L8: $"count", L9: 0]
Call stack: [0, -1]
TRACE:
34365: ListAppend (L15[] = L17)
34366: Pop (2)
34367: BoxString (L14)
34368: BoxList (L15)
34369: MapSet (L5[L14] = L15)
34370: Pop (10)
34371: BoxMap (L5)  <<< EXCEPTION
34372: PushArg (L4)
34373: PushArg (L5)
34374: InvokeDynamic (L3.pushNamed)


When the exception was thrown, this was the stack
```

**Reply by ethanblake4** [reply]
Fixed in v0.5.2

**Comment by zmtzawqlp** [comment]
still has error when try to compile with dart_eval
``` dart
              NavigatorState navigatorState = Navigator.of(context);
              navigatorState.pushNamed('PageA');
```

**Reply by ethanblake4** [reply]
Can you try just doing
```dart
Navigator.of(context).pushNamed('PageA');
```

**Reply by zmtzawqlp** [reply]
when click button，still has error from
runtime

**Reply by ethanblake4** [reply]
Can you put the whole code? because that works fine for me.

I can reproduce the other errors though will fix shortly

**Reply by zmtzawqlp** [reply]
[demo.zip](https://github.com/ethanblake4/flutter_eval/files/9770068/demo.zip)
you can check it when you have a time. it seems it's late at your side.

**Reply by ethanblake4** [reply]
Hmm, I ran it and it works fine. Maybe it's just the issue that you haven't defined any navigation routes?

Anyway, published v0.5.2 which fixes the other issues. Not too late at my side yet :)

**Comment by zmtzawqlp** [comment]
[demo.zip](https://github.com/ethanblake4/flutter_eval/files/9771542/demo.zip)

work fine. but i found error when i try to add auto generate route helper class

``` dart
// GENERATED CODE - DO NOT MODIFY MANUALLY
// **************************************************************************
// Auto generated by https://github.com/fluttercandies/ff_annotation_route
// **************************************************************************
// fast mode: false
// version: 10.0.6
// **************************************************************************
// ignore_for_file: prefer_const_literals_to_create_immutables,unused_local_variable,unused_import,unnecessary_import,unused_shown_name,implementation_imports,duplicate_import
import 'package:flutter/foundation.dart';
import 'package:flutter/src/foundation/key.dart'
    as autoimport61a64860a48a0a727515bc4ec00b2fff;

const List<String> routeNames = <String>[
  'HomePage',
  'PageA',
];

class Routes {
  const Routes._();

  /// 'HomePage'
  ///
  /// [name] : 'HomePage'
  ///
  /// [exts] : {'library': 'package:runtime_widgets/src/pages/home_page.dart', 'function': 'HomePageWidget.'}
  static const String homePage = 'HomePage';

  /// 'PageA'
  ///
  /// [name] : 'PageA'
  ///
  /// [constructors] :
  ///
  /// PageA({Key? key, required int count})
  ///
  /// [exts] : {'library': 'package:runtime_widgets/src/pages/page_a.dart', 'function': 'PageAWidget.'}
  static const _PageA pageA = _PageA();
}

class _PageA {
  const _PageA();

  String get name => 'PageA';

  Map<String, dynamic> d(
          {autoimport61a64860a48a0a727515bc4ec00b2fff.Key? key,
          required int count}) =>
      <String, dynamic>{
        'key': key,
        'count': count,
        'constructorName': '',
        'argumentNames': ['key', 'count'],
      };

  @override
  String toString() => name;
}
```

**Reply by ethanblake4** [reply]
Fixed with dart_eval v0.5.3 (no flutter_eval changes needed)

**Reply by nicktt19** [reply]
hello, i am following the readme to build the example, and get a new CompileError, thanks.

![image](https://user-images.githubusercontent.com/106294970/204701347-643b69a1-f84a-4b4e-958e-7256d4e2a3a7.png)

**Reply by ethanblake4** [reply]
@nicktt19 please create a new discussion so I can help you. That message is unrelated to this discussion with @zmtzawqlp

**Reply by nicktt19** [reply]
ok,thanks.
